### PR TITLE
Integration test for SetMessages (sending message)

### PR DESCRIPTION
**Pull request by fvn-linagora**
add more robust assertions (awaitility) to SetMessages method
add an integration test for sending an email involving 2 users on same domain

### CODE REVIEW

**Comment by rouazana** [comment]
:+1:

File: server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -91,32 +99,39 @@ public void setup() throws Exception {
...
         RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
 
-        String domain = "domain.tld";
-        username = "username@" + domain;
+        username = "username@" + USERS_DOMAIN;
         String password = "password";
-        jmapServer.serverProbe().addDomain(domain);
+        jmapServer.serverProbe().addDomain(USERS_DOMAIN);
         jmapServer.serverProbe().addUser(username, password);
         jmapServer.serverProbe().createMailbox("#private", "username", "inbox");
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:106]: username without quotes

File: server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java [hunk 2]
```diff
@@ -68,12 +72,16 @@
...
+    public static final String SECOND_NAME = "[1][0]";
+    public static final String SECOND_ARGUMENTS = "[1][1]";
+    public static final String USERS_DOMAIN = "domain.tld";
 
     private final TemporaryFolder temporaryFolder = new TemporaryFolder();
     private final EmbeddedElasticSearch embeddedElasticSearch = new EmbeddedElasticSearch(temporaryFolder);
     private final EmbeddedCassandra cassandra = EmbeddedCassandra.createStartServer();
     private final JmapServer jmapServer = jmapServer(temporaryFolder, embeddedElasticSearch, cassandra);
-    private String outboxId;
+    private String sentMailboxId;
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:83]: why a field?

File: server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java [hunk 3]
```diff
@@ -897,4 +1014,80 @@ public void setMessagesShouldRejectWhenSendingMessageHasMissingSubject() {
...
+                    .body("[[\"getMessageList\", {\"fetchMessages\":true, \"fetchMessageProperties\": [\"mailboxIds\"]}, \"#0\"]]")
+            .when()
+                    .post("/jmap")
+            .then()
+                    .statusCode(200)
+                    .body(NAME, equalTo("messageList"))
+                    .body(ARGUMENTS + ".messageIds", hasSize(1))
+                    .body(SECOND_NAME, equalTo("messages"))
+                    .body(SECOND_ARGUMENTS + ".list", hasSize(1))
+                    // .body(SECOND_ARGUMENTS + ".list[0].mailboxIds", contains(recipientInboxId))
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:1085]: uncomment?

Comment (fvn-linagora) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:1085]: removed assertions on messages (SECOND_ARGUMENTS).
Had issues with filtering and fetching in getMessageList, thus needed to revert to "empty query" for asserting recipient user had received a message.

File: server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java (hunk 1 above)
Comment (fvn-linagora) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:106]: fixed, seems like it had been there for a while though.

File: server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java (hunk 2 above)
Comment (fvn-linagora) [inline server/protocols/jmap-integration-testing/src/test/java/org/apache/james/jmap/methods/SetMessagesMethodTest.java:83]: removed

**Comment by fvn-linagora** [comment]
following up in https://github.com/linagora/james-project/pull/174

**Pull request closed without merging**